src/lib/crypto/secp256k1.ts

```diff
@@ -6,6 +6,11 @@ import {
   Secp256k1Wasm
 } from '../bin/bin';
 
+export interface RecoverableSignature {
+  readonly recovery: number;
+  readonly signature: Uint8Array;
```

Comment:
This is a bit arcane (sorry) – we should remove the `readonly` modifiers here. Internally, we want to accept everything as `readonly` to avoid mutating inputs and so that consumers can easily pass us `readonly` properties without type-checking failing. But we want to return everything *without* the `readonly` modifier to allow consumers the maximum flexibility when using the library. If we return consumers a `readonly` property, type-checking will force them to copy the value to before modifying it (useful if a library actually needs something to be readonly, but in our case, we're done with the result when we give it back to the consumer).

`tslint` will complain – so you'll need to add the line:
```
// tslint:disable-next-line:readonly-keyword
```
before each line.